products/views.py

```diff
@@ -1,3 +1,62 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http import JsonResponse
+from django.views import View
+from django.db.models import Q, Sum
+
+from products.models import Product
+
+class ProductsView(View):
+    def get(self,request):
+        category        = request.GET.get('category',None)
+        sub_category    = request.GET.get('sub_category',None)
+        detail_category = request.GET.get('detail_category',None)
+        color           = request.GET.getlist('color',None)
+        sorting         = request.GET.get('sorting',None)
+        search          = request.GET.get('KeyWord',None)
+        limit           = request.GET.get('limit',None)
+        offset          = int(request.GET.get('offset',0))
+    
+        filters = Q()
+
+        if category :
+            filters &= Q(detail_category__sub_category__category= category)
+
+        if sub_category :
+            filters &= Q(detail_category__sub_category__option = sub_category)
+
+        if detail_category :
+            filters &= Q(detail_category__detail = detail_category)
+        
+        if color :
+            filters &= Q(productoption__color__in = color)
+
+        if search :
+            filters = (Q(name__icontains = search)|
+                Q(detail_category__detail__icontains = search)|
+                Q(detail_category__sub_category__option__icontains = search)|
+                Q(detail_category__sub_category__category__name__icontains = search)
+                )
+        
+        sort_name = {
+            'new'        : '-create_at',
+            'bestseller' : '-sales',
+        }
+
+        if limit:
+            limit = offset + int(limit)
+
+        products = Product.objects.filter(filters).annotate(sales = Sum('productoption__sales'))\
+            .order_by(sort_name.get(sorting,'id'))[offset:limit]
```

Comment:
```suggestion
        products = Product.objects.filter(filters)\
                                  .annotate(sales = Sum('productoption__sales'))\
                                  .order_by(sort_name.get(sorting,'id'))[offset:limit]
```
다음과 같이 수정하는 것이 가독성에 더 좋을 것  같습니다.

Reply:
넵 수정 하겠습니다.